### PR TITLE
Replace deprecated nose style setup/teardown with autouse fixtures

### PR DESCRIPTION
**Pull request by lagru**
## Description

Follow-up to https://github.com/scikit-image/scikit-image/pull/7340. 

Pytest has deprecated nose style setup and teardown since 7.2. 

https://docs.pytest.org/en/latest/deprecations.html#nose-deprecation

## Checklist

- A descriptive but concise pull request title
- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [Unit tests](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- A gallery example in `./doc/examples` for new features
- [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

## Release note

We use [changelist](https://github.com/scientific-python/changelist) to
compile each pull request into an item of the release notes. Please refer to
the [instructions](https://scikit-image.org/docs/stable/development/contribute.html#documenting-changes)
and [past release notes](https://scikit-image.org/docs/stable/release_notes/index.html)
for guidance and examples.

```release-note
...
```

### CODE REVIEW

**Review by jarrodmillman — APPROVED** [review]

File: skimage/io/tests/test_imread.py [hunk 1]
```diff
@@ -12,16 +12,16 @@
...
-from pytest import importorskip
+import pytest
 
-importorskip('imread')
+pytest.importorskip('imread')
 
 
-def setup():
+@pytest.fixture(autouse=True)
+def _use_imread_plugin():
```
Comment (mkcor) [inline skimage/io/tests/test_imread.py:21]: Disclaimer: non-expert here :wink: Just curious, from reading the [documentation you linked to](https://docs.pytest.org/en/latest/deprecations.html#nose-deprecation), I was expecting some `def setup()` to be replaced by `def setup_method()` (while the fixture is recommended for replacing code using `@with_setup`)...(?)

**Review by mkcor — COMMENTED** [review]
I can see the changes are consistent, but I'm *not* knowledgeable enough to approve!

File: skimage/io/tests/test_imread.py (hunk 1 above)
Comment (lagru) [reply skimage/io/tests/test_imread.py:21]: Ah, I didn't know that pytest supports another approach to achieve this. :sweat_smile: Though, in this case it should be [`setup_module` and `teardown_module`](https://docs.pytest.org/en/latest/how-to/xunit_setup.html#module-level-setup-teardown).

Looks like both approaches are valid but I feel like the fixture based one may the better known approach and is also the one recommended by pytest. And that I feel more comfortable with the fixture based approach may also be a factor. ^^

**Comment by lagru** [comment]
Merging since it seems there's a silent consensus and at least 2 approving maintainers. :)

**Pull request merged**